packages/tools/src/tools/displayTools/Contour/updateContourSets.ts

```diff
@@ -0,0 +1,111 @@
+import { cache, Types } from '@cornerstonejs/core';
+import vtkDataArray from '@kitware/vtk.js/Common/Core/DataArray';
+import vtkAppendPolyData from '@kitware/vtk.js/Filters/General/AppendPolyData';
+import vtkActor from '@kitware/vtk.js/Rendering/Core/Actor';
+
+import {
+  SegmentationRepresentationConfig,
+  ToolGroupSpecificContourRepresentation,
+} from '../../../types';
+import { getConfigCache, setConfigCache } from './contourConfigCache';
+import { getPolyData } from './utils';
+
+export function updateContourSets(
+  viewport: Types.IVolumeViewport,
+  geometryIds: string[],
+  contourRepresentation: ToolGroupSpecificContourRepresentation,
+  contourRepresentationConfig: SegmentationRepresentationConfig,
+  contourActorUID: string
+) {
+  const { segmentationRepresentationUID, segmentsHidden } =
+    contourRepresentation;
+  const newContourConfig = contourRepresentationConfig.representations.CONTOUR;
+  const cachedConfig = getConfigCache(segmentationRepresentationUID);
+
+  const contourSetsActor = viewport.getActor(contourActorUID);
+
+  if (!contourSetsActor) {
+    console.warn(
+      `No contour actor found for actorUID ${contourActorUID}. Skipping render.`
+    );
+    return;
+  }
+
+  const { actor } = contourSetsActor;
+
+  const newOutlineWithActive = newContourConfig.outlineWidthActive;
+
+  if (cachedConfig.outlineWidthActive !== newOutlineWithActive) {
```

Comment:
Is there a chance that `cachedConfig` might be `undefined` or `null`?

Reply:
there shouldn't be since we add first, and that caches it, but I added the checks